client/client.go

```diff
@@ -247,76 +216,154 @@ func (c *Client) update(latestRoot bool) (data.TargetFiles, error) {
+		// of root rotation. AFTER the rotation is over, we will set it to the value of the last root.
+		consistentSnapshot = nPlusOneRootMetadataSigned.ConsistentSnapshot
+		// 5.3.8 Persist root metadata. The client MUST write the file to non-volatile storage as FILENAME.EXT (e.g. root.json).
+		// NOTE: Internally, setMeta stores metadata in LevelDB in a persistent manner.
+		if err := c.local.SetMeta("root.json", nPlusOneRootMetadata); err != nil {
+			return err
+		}
+		nRootMetadata = nPlusOneRootMetadata
+		// 5.3.9 Repeat steps 5.3.2 to 5.3.9
+
+	} // End of the for loop.
+
+	// 5.3.10 Check for a freeze attack.
+	// NOTE: this will check for any, including freeze, attack.
+	if err := c.loadAndVerifyLocalRootMeta(); err != nil {
+		return err
 	}
-	if err := c.local.SetMeta("root.json", rootJSON); err != nil {
-		return nil, err
+
+	// 5.3.11 If the timestamp and / or snapshot keys have been rotated,
+	// then delete the trusted timestamp and snapshot metadata files.
+	for topLevelRolename := range nonRootManifests {
+		if !reflect.DeepEqual(
+			nonRootManifests[topLevelRolename],
+			getKeyIDs(topLevelRolename)) {
+			c.local.SetMeta(topLevelRolename, json.RawMessage{})
+		}
 	}
-	return c.update(true)
+
+	// 5.3.12 Set whether consistent snapshots are used as per the trusted root metadata file.
+	c.consistentSnapshot = consistentSnapshot
+	return nil
 }
 
 // getLocalMeta decodes and verifies metadata from local storage.
-//
 // The verification of local files is purely for consistency, if an attacker
 // has compromised the local storage, there is no guarantee it can be trusted.
 func (c *Client) getLocalMeta() error {
-	meta, err := c.local.GetMeta()
-	if err != nil {
+	if err := c.loadAndVerifyLocalRootMeta(); err != nil {
 		return err
 	}
 
-	if rootJSON, ok := meta["root.json"]; ok {
-		// unmarshal root.json without verifying as we need the root
-		// keys first
-		s := &data.Signed{}
-		if err := json.Unmarshal(rootJSON, s); err != nil {
-			return err
-		}
-		root := &data.Root{}
-		if err := json.Unmarshal(s.Signed, root); err != nil {
-			return err
-		}
-		c.db = verify.NewDB()
-		for id, k := range root.Keys {
-			if err := c.db.AddKey(id, k); err != nil {
-				// TUF is considering in TAP-12 removing the
-				// requirement that the keyid hash algorithm be derived
-				// from the public key. So to be forwards compatible,
-				// we ignore `ErrWrongID` errors.
-				//
-				// TAP-12: https://github.com/theupdateframework/taps/blob/master/tap12.md
-				if _, ok := err.(verify.ErrWrongID); !ok {
-					return err
-				}
-			}
-		}
-		for name, role := range root.Roles {
-			if err := c.db.AddRole(name, role); err != nil {
-				return err
-			}
-		}
-		if err := c.db.Verify(s, "root", 0); err != nil {
-			return err
-		}
-		c.consistentSnapshot = root.ConsistentSnapshot
-	} else {
-		return ErrNoRootKeys
+	meta, err := c.local.GetMeta()
+	if err != nil {
+		return nil
 	}
 
 	if snapshotJSON, ok := meta["snapshot.json"]; ok {
```

Comment:
The rest of this logic is irrelevant to this PR, but the logic is not even correct. First you should verify timestamp against root, then snapshot against timestamp, then targets against snapshot.